src/sail_json_backend/json.ml

```diff
@@ -398,14 +398,77 @@ let remove_identity_funcs op =
       else op
   ) op identity_funcs
 
+let optional_operand k op =
+  let pattern = Str.regexp ("maybe_[A-Za-z_]+\\(" ^ "(" ^ op ^ ")" ^ "\\)") in
+  match Hashtbl.find_opt assembly_clean k with
+  | Some op_list ->
+      let captured_operand =
+        List.find_map (fun s ->
+          if Str.string_match pattern s 0 then
+            let captured_group = Str.matched_group 0 s in
+            let optional_operand_parts = String.split_on_char '(' captured_group in
+            let captured_op = List.hd optional_operand_parts in
+            debug_print ("Captured operand: " ^ captured_op);
+            Some captured_op
+          else
+            None
+        ) op_list
+      in
+      (match captured_operand with
+      | Some op -> Some true, op
+      | None -> None, "")
+  | None -> None, ""
+
+let default_operand mappings key =
+  if Hashtbl.mem mappings key then
+    let values = Hashtbl.find_all mappings key in
+    let zero_bit_pattern s = Str.string_match (Str.regexp "0b0*$") s 0 in
```

Comment:
This is also a bit hacky, depending on a certain value in the mappings.
Could you add an explanatory comment here as well?

Reply:
added the come. Let me know if this needs more detailing.